config_test.go

```diff
@@ -145,3 +146,106 @@ func TestConfigShellCmdsString(t *testing.T) {
 		})
 	}
 }
+
+func ExampleConfigShellCmds_AppendScript() {
+	shCmds := ConfigShellCmds{}
+	shCmds.AppendScript(`
+		# This script will be unindented by the number of leading tabs
+		# on the first line.
+		if true; then
+			echo "this is always printed"
+		fi`,
+	)
+	b, _ := json.MarshalIndent(&shCmds, "", "  ")
+	fmt.Println(string(b))
+
+	// Output:
+	// [
+	//   "# This script will be unindented by the number of leading tabs",
+	//   "# on the first line.",
+	//   "if true; then",
+	//   "\techo \"this is always printed\"",
+	//   "fi"
+	// ]
+}
+
+func TestAppendScript(t *testing.T) {
+	tests := []struct {
+		name     string
+		script   string
+		wantCmds []string
+	}{
+		{
+			name:     "Empty",
+			script:   "",
+			wantCmds: nil,
+		},
+		{
+			name:     "OnlySpaces",
+			script:   " ",
+			wantCmds: nil,
+		},
+		{
+			name:     "Only newlines",
+			script:   "\r\n",
+			wantCmds: nil,
+		},
+		{
+			name:     "Simple",
+			script:   "echo test",
+			wantCmds: []string{"echo test"},
+		},
+		{
+			name:     "LeadingNewline",
+			script:   "\necho test",
+			wantCmds: []string{"echo test"},
+		},
+		{
+			name:     "LeadingNewlineAndSpace",
+			script:   "\n    echo test",
+			wantCmds: []string{"    echo test"},
```

Comment:
why trim leading tabs but not leading spaces?

Reply:
Updated to also trim leading spaces from the script.

Reply:
It only unindents by tabs because Go source is always indented with tabs. I figured there wouldn't really be a case where we'd want to indent using tabs.